Services/EmployeeSalesService.cs

```diff
@@ -0,0 +1,47 @@
+using System.Data;
+using Microsoft.AspNetCore.Mvc;
+
+public class EmployeeSalesService
+{
+    private DataClient _connection;
+    public EmployeeSalesService(DataClient connection){
+        _connection = connection;
+    }
+    public List<EmployeeSalesModel>? GetsalesByNameAndyear(string name, DateTime year){
+        try{
+            var result = _connection.GetResultsFromQuery<EmployeeSalesModel>(
+                "SELECT soh.SalesOrderID, soh.OrderDate, soh.DueDate, soh.ShipDate, soh.Status, "+
+                "sod.SalesOrderDetailID, soh.CustomerID, sp.SalesYTD, sp.SalesLastYear, " +
+                "pe.FirstName, pe.MiddleName, pe.LastName " +
+                "FROM Sales.SalesOrderHeader soh " +
+                "INNER JOIN Sales.SalesOrderDetail sod ON soh.SalesOrderID = sod.SalesOrderID "+
+                "LEFT JOIN Sales.SalesPerson sp ON soh.SalesPersonID = sp.BusinessEntityID "+
+                "INNER JOIN HumanResources.Employee hr ON sp.BusinessEntityID = hr.BusinessEntityID " +
+                "INNER JOIN Person.Person pe ON hr.BusinessEntityID =pe.BusinessEntityID " +
+                "WHERE " + 
+                $"    ('{name}' ='' OR '{name}' IS NULL OR CONCAT(pe.FirstName, ' ', pe.MiddleName, ' ', pe.LastName) LIKE '%{name}%') " +
+                $"    AND " +
+                $"    ('{ year }' = '' OR '{year}' IS NULL OR soh.OrderDate = '{year}')", Map);
+            return result;
+        }catch (Exception ex){
+            Console.WriteLine($"JustError: {ex.Message}");
+        }
+        return null;
+    }
+
+
+    public EmployeeSalesModel Map(IDataRecord record){
+        EmployeeSalesModel Emple = new EmployeeSalesModel();
```

Comment:
First, the names of internal variables should be better using camelCase. This means starting with a lowercase letter, and if there are combinations of words, you can use uppercase to separate them, e.g., 'employeeData'. In this case, 'employee' is recommended. Using 'Emple' can lead to misunderstandings when another developer begins to work on this method. You can find more information on this topic here: [C# Coding Standards and Naming Conventions](https://www.c-sharpcorner.com/UploadFile/8a67c0/C-Sharp-coding-standards-and-naming-conventions/)